aws/resource_aws_glue_catalog_crawler.go

```diff
@@ -0,0 +1,411 @@
+	}
+	return schemaPolicy
+}
+
+func createCrawlerTargets(d *schema.ResourceData) (*glue.CrawlerTargets, error) {
+	crawlerTargets := &glue.CrawlerTargets{}
+
+	jdbcTargets, jdbcTargetsOk := d.GetOk("jdbc_target")
+	s3Targets, s3TargetsOk := d.GetOk("s3_target")
+	if !jdbcTargetsOk && !s3TargetsOk {
+		return nil, fmt.Errorf("jdbc targets or s3 targets configuration is required")
+	}
+
+	log.Print("[DEBUG] Creating crawler target")
+	crawlerTargets.S3Targets = expandS3Targets(s3Targets.(*schema.Set).List())
+	crawlerTargets.JdbcTargets = expandJdbcTargets(jdbcTargets.(*schema.Set).List())
+
+	return crawlerTargets, nil
+}
+
+func expandS3Targets(targets []interface{}) []*glue.S3Target {
+	if len(targets) < 1 {
+		return []*glue.S3Target{}
+	}
+
+	perms := make([]*glue.S3Target, len(targets), len(targets))
+	for i, rawCfg := range targets {
+		cfg := rawCfg.(map[string]interface{})
+		perms[i] = expandS3Target(cfg)
+	}
+	return perms
+}
+
+func expandS3Target(cfg map[string]interface{}) *glue.S3Target {
+	target := &glue.S3Target{
+		Path: aws.String(cfg["path"].(string)),
+	}
+
+	if exclusions, ok := cfg["exclusions"]; ok {
+		target.Exclusions = expandStringList(exclusions.([]interface{}))
+	}
+	return target
+}
+
+func expandJdbcTargets(targets []interface{}) []*glue.JdbcTarget {
+	if len(targets) < 1 {
+		return []*glue.JdbcTarget{}
+	}
+
+	perms := make([]*glue.JdbcTarget, len(targets), len(targets))
+	for i, rawCfg := range targets {
+		cfg := rawCfg.(map[string]interface{})
+		perms[i] = expandJdbcTarget(cfg)
+	}
+	return perms
+}
+
+func expandJdbcTarget(cfg map[string]interface{}) *glue.JdbcTarget {
+	target := &glue.JdbcTarget{
+		Path:           aws.String(cfg["path"].(string)),
+		ConnectionName: aws.String(cfg["connection_name"].(string)),
+	}
+
+	if exclusions, ok := cfg["exclusions"]; ok {
+		target.Exclusions = expandStringList(exclusions.([]interface{}))
+	}
+	return target
+}
+
+func resourceAwsGlueCatalogCrawlerUpdate(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	crawlerInput, err := createCrawlerInput(name, d)
+	if err != nil {
+		return err
+	}
+
+	crawlerUpdateInput := glue.UpdateCrawlerInput(*crawlerInput)
+	if _, err := glueConn.UpdateCrawler(&crawlerUpdateInput); err != nil {
+		return err
+	}
+
+	return resourceAwsGlueCatalogCrawlerRead(d, meta)
+}
+
+func resourceAwsGlueCatalogCrawlerRead(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	input := &glue.GetCrawlerInput{
+		Name: aws.String(name),
+	}
+
+	crawlerOutput, err := glueConn.GetCrawler(input)
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			log.Printf("[WARN] Glue Crawler (%s) not found, removing from state", d.Id())
+			d.SetId("")
+		}
+
+		return fmt.Errorf("error reading Glue crawler: %s", err.Error())
+	}
+
```

Comment:
To prevent potential panics, we should check that `crawlerOutput.Crawler` is not `nil` beforehand:

```go
if crawlerOutput.Crawler == nil {
  log.Printf("[WARN] Glue Crawler (%s) not found, removing from state", d.Id())
  d.SetId("")
  return nil
}
```

Reply:
Done